src/decoder.ts

```diff
@@ -293,6 +293,43 @@ export class Decoder<A> {
     });
   }
 
+  static strictObject<A>(decoders?: DecoderObject<A>) {
+    const decoderKeys = isJsonObject(decoders) ? Object.keys(decoders) : [];
+    return new Decoder((json: unknown) => {
+      const invalidKeys = isJsonObject(json)
+        ? Object.keys(json).filter((key: string) => !decoderKeys.some((validKey: string) => key === validKey))
+        : [];
+
+      if (invalidKeys.length > 0) {
+        return Result.err({message: `the following keys are present but not expected: ${invalidKeys.join(", ")}`});
+      }
```

Comment:
it might be better to represent this decoder as the intersection of the standard `object` decoder and another no extra keys decoder. Avoid copying the contents of that decoder into this one.